### ISSUE TITLE
Feature Request: Generate separate markdown files per entry

### ISSUE THREAD

**Issue by motordigital** [contentful/jekyll-contentful-data-import#25]
The existing importer is great and works fine as it is. But for creating a more flexible website based on Contentful data, some features are not possible by a single large data yaml file. Specially it is a URL generation by Jekyll (without special plugins).

Therefor I have something like that in mind:

* configure per content type if it should be rendered in a large yaml data file or in separate markdown files
* for markdownfiles define
** location to same the files, e.g. /teammembers/_posts/
** Contentful field for usage as markdown content
** Contentful field for the date within the markdown filename
** template key/value pairs for yaml (e.g. Jekyll typical layout property)
** simplification for references used as Jekyll category and tag (e.g. by a configuration of Contenful fields used for tags and categories and than reduce the reference entries to eliminate ids)

**Comment by dlitvakb** [comment]
We're working on something similar to this, you can follow progress here: https://github.com/contentful/jekyll-contentful-data-import/pull/23

Once this is done we can look into tackling the remaining points here.

**Comment by felixjung** [comment]
@motordigital See my comments regarding your first request in #23. I no longer think it's necessary to add such a configuration flag.

With respect to your second comment, I don't think the approach of creating actual files outside of `_data` would be a very scalable/flexible approach. It appears very blog/collection specific to me. What if, for example, you wanted to create a single page for each entry of a given content type (i.e. something that becomes like `my-entry/index.html` inside `_site`)?

Having looked at the [Jekyll docs for Generators](http://jekyllrb.com/docs/plugins/#generators), I found it quite easy to just generate blog posts or other arbitrary pages on the fly during the build. I'm not entirely done with this current project, but I'm confident we will be able to have a static Jekyll site where partner langing pages, the blog, our press coverage pages etc. are all pulled from Contentful and then rendered to static html by Jekyll.

Yes, as you mention, this means additional plugins, but tiny ones. And since you're already running the Contentful plugin, I can't see how that's an issue other than having to put in a bit more work (Github pages is out of the question at this point). In the end, this approach gets you small composable plugins  rather than one monolithic Contentful plugin taking care of all kinds of rendering tasks. That's why we went headless in the first place, no? 😉
